internal/app/readycheck.go

```diff
@@ -1,13 +1,157 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package app
 
-import "net/http"
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
 
-func (a *Application) readinessHandler(w http.ResponseWriter, r *http.Request) {
+	"github.com/gardener/etcd-wrapper/internal/util"
 
-}
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+const (
+	// ReadyServerPort is the port number for the server that serves the readiness probe
+	ReadyServerPort       = int64(9095)
```

Comment:
Any particular reason to not use the default port of `8080` for readiness?

Reply:
No specific reason per se. We wanted to use an unused port so that it would not overlap with anything that we might build later on

Reply:
Port number of 8080 is fine. Since every pod has an isolated network namespace this should be fine. There is no need to use 9095. Care should be taken that if there are pods having 2 containers and both expose http(s) endpoints via a server then they should use different ports as all containers within a pod share the same network namespace.

Reply:
Ok